BitwardenShared/Core/Auth/Services/AuthService.swift

```diff
@@ -636,11 +658,14 @@ class DefaultAuthService: AuthService { // swiftlint:disable:this type_body_leng
         resendEmailModel = nil
 
         if try await canUnlockWithDeviceKey(response) {
-            return nil
+            return .deviceKey
         }
 
-        // Return the account if the vault still needs to be unlocked.
-        return try await stateService.getActiveAccount()
+        if try await canUnlockWithKeyConnectorKey(response) {
+            return .keyConnector
+        }
+
+        return try await .masterPassword(stateService.getActiveAccount())
```

Comment:
🎨 I see this is repeated on `func loginWithSingleSignOn(code: String, email: String) async throws -> LoginUnlockMethod`. Could we extract this to its own func to be reused? 
🎨 Additionally, it seems Codecov is not finding tests for the `.deviceKey` path and I'm not finding them neither, could a test be added for that case?

Reply:
Good callout, I'll extract this and add a test for the device key flow!